sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpressions.scala

```diff
@@ -295,85 +272,72 @@ trait GpuTernaryExpression extends TernaryExpression with GpuExpression {
-          doColumnar(v0, v1, v2)
-        case (v0, v1: GpuColumnVector, v2: GpuColumnVector) =>
-          val scalar0 = GpuScalar.from(v0, children(0).dataType)
-          try {
-            GpuColumnVector.from(doColumnar(scalar0, v1, v2), dataType)
-          } finally {
-            scalar0.close()
-          }
-        case (v0: GpuColumnVector, v1, v2: GpuColumnVector) =>
-          val scalar1 = GpuScalar.from(v1, children(1).dataType)
-          try {
-            GpuColumnVector.from(doColumnar(v0, scalar1, v2), dataType)
-          } finally {
-            scalar1.close()
-          }
-        case (v0: GpuColumnVector, v1: GpuColumnVector, v2) =>
-          val scalar2 = GpuScalar.from(v2, children(2).dataType)
-          try {
-            GpuColumnVector.from(doColumnar(v0, v1, scalar2), dataType)
-          } finally {
-            scalar2.close()
-          }
-        case (v0, v1, v2: GpuColumnVector) =>
-          val scalar0 = GpuScalar.from(v0, children(0).dataType)
-          val scalar1 = GpuScalar.from(v1, children(1).dataType)
-          try {
-            GpuColumnVector.from(doColumnar(scalar0, scalar1, v2), dataType)
-          } finally {
-            scalar0.close()
-            scalar1.close()
-          }
-        case (v0, v1: GpuColumnVector, v2) =>
-          val scalar0 = GpuScalar.from(v0, children(0).dataType)
-          val scalar2 = GpuScalar.from(v2, children(2).dataType)
-          try {
-            GpuColumnVector.from(doColumnar(scalar0, v1, scalar2), dataType)
-          } finally {
-            scalar0.close()
-            scalar2.close()
-          }
-        case (v0: GpuColumnVector, v1, v2) =>
-          val scalar1 = GpuScalar.from(v1, children(1).dataType)
-          val scalar2 = GpuScalar.from(v2, children(2).dataType)
-          try {
-            GpuColumnVector.from(doColumnar(v0, scalar1, scalar2), dataType)
-          } finally {
-            scalar1.close()
-            scalar2.close()
-          }
-        case (v0, v1, v2) if v0 != null && v1 != null && v2 != null =>
-          withResource(GpuScalar.from(v0, children(0).dataType)) { v0Scalar =>
-            withResource(GpuScalar.from(v1, children(1).dataType)) { v1Scalar =>
-              withResource(GpuScalar.from(v2, children(2).dataType)) { v2Scalar =>
-                GpuColumnVector.from(doColumnar(batch.numRows(), v0Scalar, v1Scalar, v2Scalar),
-                  dataType)
+    withResourceIfAllowed(children(0).columnarEval(batch)) { val0 =>
+      withResourceIfAllowed(children(1).columnarEval(batch)) { val1 =>
+        withResourceIfAllowed(children(2).columnarEval(batch)) { val2 =>
+          (val0, val1, val2) match {
+            case (v0: GpuColumnVector, v1: GpuColumnVector, v2: GpuColumnVector) =>
+              doColumnar(v0, v1, v2)
+            case (v0, v1: GpuColumnVector, v2: GpuColumnVector) =>
+              val scalar0 = GpuScalar.from(v0, children(0).dataType)
```

Comment:
Why was this, and all the other scalar args in here not update to `withResource` too?

Reply:
Good point that it should be updated, will add it. I missed it because I was actually addressing the "instanceOf" warnings, and scalars are not affected.